### PR TITLE
Reindex post if post meta is in allow list

### PR DESCRIPTION
**Pull request by brandon-m-skinner**
## Description

Depends upon https://github.com/Automattic/ElasticPress/pull/44

Since the post meta allow list has been added, posts should not be re-indexed if meta not on that list are changed.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [x] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

1. `./bin/phpunit-docker.sh tests/search/test-class-search.php`

------

1. Apply PR. Ensure https://github.com/Automattic/ElasticPress/pull/44 is applied as well.
2. Ensure VIP Search in enabled and you have a properly consistent index.
3. Add some post meta to a post that isn't on the allow list. The post should not have been re-indexed
4. Change the value of that meta. The post should not have been re-indexed.
5. Delete the meta. The post should not have been re-indexed.
6. Add some post meta to a post that is on the allow list. The post should be re-indexed.
7. Change the value of that meta. The post should be re-indexed.
8. Delete the meta. The post should be re-indexed.

### CODE REVIEW

**Comment by brandon-m-skinner** [comment]
I see. So when a filter/action is hooked, it never unhooks unless that's explicitly done. Gotcha.

File: search/includes/classes/class-search.php [hunk 1]
```diff
@@ -1309,6 +1312,71 @@ public function limit_field_limit( $field_limit ) {
...
+				return $skip_sync;
+			}
+		}
+
+		// If post is invalid, respect current sync value
+		if ( is_null( $post ) ) {
+			return $skip_sync;
+		}
+
+		$post_meta_allow_list = \apply_filters( 'vip_search_post_meta_allow_list', self::POST_META_DEFAULT_ALLOW_LIST, $post );
```
Comment (nickdaugherty) [inline search/includes/classes/class-search.php:1343]: Would it make sense to wrap this `apply_filters()` call into a reusable function in the Search class, to encapsulate (and de-duplicate) the logic?

Comment (nickdaugherty) [reply search/includes/classes/class-search.php:1343]: Better yet, all the code below should be de-duplicated too. If we can extract this out into something that both this function and `filter__ep_prepare_meta_data()` can use, it'll keep the core meta allow list logic DRY.

File: tests/search/test-class-search.php [hunk 2]
```diff
@@ -1923,6 +1923,66 @@ function() {
...
+					'random_key',
+				);
+			}
+		);
+
+		$es = \Automattic\VIP\Search\Search::instance();
+
+		$this->assertTrue( $es->filter__ep_skip_post_meta_sync( true, $post, 40, 'random_key', 'random_value' ) );
+	}
+
```
Comment (nickdaugherty) [inline tests/search/test-class-search.php:1985]: These unit tests are good for verifying the behavior of `filter__ep_skip_post_meta_sync()`, but we should also verify that our actual filter is being hooked and run correctly. So a similar test, but one that just runs `apply_filters( 'ep_skip_post_meta_sync', ... )` - that'll prove that we are correctly hooking into EP.

**Comment by brandon-m-skinner** [comment]
r1029-stacks

**Pull request merged**